src/components/modals/RunActionModal.svelte

```diff
@@ -0,0 +1,75 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte';
+  import type { ActionDefinition } from '../../types/actions';
+  import type { User } from '../../types/app';
+  import type { ArgumentsMap, FormParameter } from '../../types/parameter';
+  import { valueSchemaRecordToParametersMap } from '../../utilities/actions';
+  import effects from '../../utilities/effects';
+  import { getArguments, getFormParameters } from '../../utilities/parameters';
+  import { showFailureToast } from '../../utilities/toast';
+  import Parameters from '../parameters/Parameters.svelte';
+  import Modal from './Modal.svelte';
+  import ModalContent from './ModalContent.svelte';
+  import ModalFooter from './ModalFooter.svelte';
+  import ModalHeader from './ModalHeader.svelte';
+
+  export let actionDefinition: ActionDefinition;
+  export let user: User | null;
+
+  let running: boolean = false;
+  let argumentsMap: ArgumentsMap = {};
+
+  const dispatch = createEventDispatcher<{
+    close: void;
+    complete: { actionRunId: number | null };
+  }>();
+
+  async function run() {
+    try {
+      running = true;
+      const actionRunId = await effects.createActionRun(
+        actionDefinition.id,
+        argumentsMap,
+        actionDefinition.settings,
+        user,
+      );
```

Comment:
Do we need to do any checking here to ensure that there isn't an existing action run even though we temporarily disable the button? Or do we not care if the user queues up multiple runs?

Reply:
I think I'd lean towards not being overly restrictive yet. Maybe we revisit this if action queuing/concurrency changes? Could imagine a long running action that reads a sequence and posts some analysis to a different server. So the user might run the analysis, tweak the sequence a bit, and run it again and compare the results.